### PR TITLE
Fixing nested remote dependencies

### PR DESCRIPTION
**Pull request by brgaulin**
Currently if you have dependencies with dependencies, they tar and folder get deleted before the files are all loaded. This is kind of a hack, but works. 

Also just some cleanup of rubocop/rubymine warnings

### CODE REVIEW

File: lib/jenkins_pipeline_builder/remote_dependencies.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def load(dependencies)
           file = entries[url]
         else
           opts = {}
-          opts = { ssl_verify_mode: OpenSSL::SSL::VERIFY_NONE } if source[:verify_ssl] == false
+          opts = { ssl_verify_mode: OpenSSL::SSL::VERIFY_NONE } unless source[:verify_ssl]
```
Comment (crimsonknave) [inline lib/jenkins_pipeline_builder/remote_dependencies.rb]: This now defaults to setting verify none when verify_ssl isn't passed. It needs to go back to the if verify_ssl == false.

**Comment by crimsonknave** [comment]
It would also be great to get a quick test that'll fail with the current code and nested depends.

File: lib/jenkins_pipeline_builder/remote_dependencies.rb (hunk 1 above)
Comment (brgaulin) [inline lib/jenkins_pipeline_builder/remote_dependencies.rb]: Ah, makes sense

**Comment by brgaulin** [comment]
will do =)

**Comment by brgaulin** [comment]
Test added, basically the file was getting deleted too early. Depending on how you nest there were weird behaviors, but this seems to clean them up =) In my case I have a dependency on a dependency that has a dependency, this test I only went two layers deep but it shows a similar issue.

**Pull request merged**